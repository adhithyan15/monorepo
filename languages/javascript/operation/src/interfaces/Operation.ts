export interface Operation<OperationReturnType> {
    getOperationName(): string;
    getResult(): OperationReturnType;
}