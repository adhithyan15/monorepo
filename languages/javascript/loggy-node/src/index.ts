import { LoggerFactory } from "../../loggy/src/interfaces/LoggerFactory";
import { LoggerFactoryImpl } from "../../loggy/src/implementations/LoggerFactoryImpl";

export function getLoggerFactory(): LoggerFactory {
  return new LoggerFactoryImpl();
}
