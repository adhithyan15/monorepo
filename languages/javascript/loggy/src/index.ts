import { LoggerFactory } from "./interfaces/LoggerFactory";
import { LoggerFactoryImpl } from "./implementations/LoggerFactoryImpl";

export function getLoggerFactory(): LoggerFactory {
    return new LoggerFactoryImpl();
}