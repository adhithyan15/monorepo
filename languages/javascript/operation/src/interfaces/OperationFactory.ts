import { Operation } from "./Operation";

export interface OperationFactory {
    createOperation<OperationReturnType>(operationName: string, callbackFunction: (operation: Operation<OperationReturnType>) => OperationReturnType): Operation<OperationReturnType>;
}