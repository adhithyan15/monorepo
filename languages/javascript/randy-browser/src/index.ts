import { RandomNumberGeneratorFactory } from "../../randy/src/interfaces/RandomNumberGeneratorFactory";
import { RandomNumberGeneratorFactoryImpl } from "../../randy/src/implementations/RandomNumberGeneratorFactoryImpl";
import { StructuredLogger } from "../../structured-logger/src/interfaces/StructuredLogger";
import { BrowserCryptoAPIWrapper } from "./implementations/BrowserCryptoAPIWrapper";

export function getRandomNumberGeneratorFactory(
  logger: StructuredLogger
): RandomNumberGeneratorFactory {
  return new RandomNumberGeneratorFactoryImpl(
    new BrowserCryptoAPIWrapper(),
    logger
  );
}
