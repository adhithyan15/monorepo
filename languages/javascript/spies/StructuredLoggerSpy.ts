import { StructuredLogger } from "../structured-logger/src/interfaces/StructuredLogger";

export interface CustomLog {
    logLevel: number,
    logMessage: string
}

export interface StructuredLogs {
    errorLogs: string[],
    warningLogs: string[],
    infoLogs: string[],
    verboseLogs: string[],
    debugLogs: string[],
    sillyLogs: string[],
    customLogs: CustomLog[]
}

export class StructuredLoggerSpy implements StructuredLogger {
  private capturedLogs: StructuredLogs;
  constructor() {
      this.capturedLogs = {
          errorLogs: [],
          warningLogs: [],
          infoLogs: [],
          verboseLogs: [],
          debugLogs: [],
          sillyLogs: [],
          customLogs: []
      };
  }
  public logError(logMessage: string): void {
    this.capturedLogs.errorLogs.push(logMessage);
  }
  public logWarning(logMessage: string): void {
      this.capturedLogs.warningLogs.push(logMessage);
  }
  public logInfo(logMessage: string): void {
      this.capturedLogs.infoLogs.push(logMessage);
  }
  public logVerbose(logMessage: string): void {
      this.capturedLogs.verboseLogs.push(logMessage);
  }
  public logDebug(logMessage: string): void {
      this.capturedLogs.debugLogs.push(logMessage);
  }
  public logSilly(logMessage: string): void {
      this.capturedLogs.sillyLogs.push(logMessage);
  }
  public logCustom(logLevel: number, logMessage: string): void {
      this.capturedLogs.customLogs.push({
          logLevel,
          logMessage
      })
  }
  public getCapturedLogs(): StructuredLogs {
      return this.capturedLogs;
  }
}