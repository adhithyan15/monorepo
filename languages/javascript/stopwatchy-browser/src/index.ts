import { StopWatchFactory } from "../../stopwatchy/src/interfaces/StopWatchFactory";
import { BrowserStopWatchFactoryImpl } from "./implementations/BrowserStopWatchFactoryImpl";
import { StructuredLogger } from "../../structured-logger/src/interfaces/StructuredLogger";

export function getStopWatchFactory(
  structuredLogger: StructuredLogger
): StopWatchFactory {
  return new BrowserStopWatchFactoryImpl(structuredLogger);
}
