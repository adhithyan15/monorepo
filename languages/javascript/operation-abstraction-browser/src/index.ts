import { OperationFactory } from "../../operation/src/interfaces/OperationFactory";
import { OperationFactoryImpl } from "../../operation/src/implementations/OperationFactoryImpl";
import { StructuredLogger } from "../../structured-logger/src/interfaces/StructuredLogger";

export function getOperationFactory(
  logger: StructuredLogger
): OperationFactory {
  return new OperationFactoryImpl(logger);
}
