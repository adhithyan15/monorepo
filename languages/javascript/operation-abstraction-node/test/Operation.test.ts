import { getOperationFactory } from "../src/index";
import { OperationFactory } from "../../operation/src/interfaces/OperationFactory";
import { Operation } from "../../operation/src/interfaces/Operation";
import { StructuredLoggerStub } from "../../stubs/StructuredLoggerStub";
import { StructuredLoggerSpy } from "../../spies/StructuredLoggerSpy";

test("calling getOperationName should return the name of the operation", () => {
  const operationFactory: OperationFactory = getOperationFactory(
    new StructuredLoggerStub()
  );
  const operationName = "HelloWorldOperation";
  const callbackFunction: () => void = () => console.log("Hello world!!");
  const operation: Operation<void> = operationFactory.createOperation(
    operationName,
    callbackFunction
  );
  expect(operation.getOperationName()).toEqual(operationName);
});

test("calling getResult should return the output of the callback function", () => {
  const operationFactory: OperationFactory = getOperationFactory(
    new StructuredLoggerStub()
  );
  const operationName = "SumOperation";
  const callbackFunction: () => number = () => 2 + 2;
  const operation: Operation<void> = operationFactory.createOperation(
    operationName,
    callbackFunction
  );
  expect(operation.getResult()).toEqual(callbackFunction());
});

test("calling getResult should return undefined for a void callback function", () => {
  const operationFactory: OperationFactory = getOperationFactory(
    new StructuredLoggerStub()
  );
  const operationName = "HelloWorldOperation";
  const callbackFunction: () => void = () => console.log("Hello World");
  const operation: Operation<void> = operationFactory.createOperation(
    operationName,
    callbackFunction
  );
  expect(operation.getResult()).toEqual(undefined);
});

test("calling addProperty from the operation callback should result in the property being added to propertybag", () => {
  const logger: StructuredLoggerSpy = new StructuredLoggerSpy();
  const operationFactory: OperationFactory = getOperationFactory(logger);
  const operationName = "TestAddPropertyOperation";
  const callbackFunction: (operation: Operation<void>) => void = (
    operation: Operation<void>
  ) => {
    operation.addProperty("TestKey", "TestValue");
  };
  const operation: Operation<void> = operationFactory.createOperation(
    operationName,
    callbackFunction
  );
  operation.getResult();
  expect(logger.getCapturedLogs()).toMatchSnapshot();
});
