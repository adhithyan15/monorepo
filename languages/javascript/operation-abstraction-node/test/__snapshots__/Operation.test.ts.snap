// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calling addProperty from the operation callback should result in the property being added to propertybag 1`] = `
Object {
  "customLogs": Array [],
  "debugLogs": Array [],
  "errorLogs": Array [],
  "infoLogs": Array [
    "{\\"operationName\\":\\"TestAddPropertyOperation\\",\\"elapsedTime\\":0.052977561950683594,\\"propertyBag\\":{\\"TestKey\\":\\"TestValue\\"}}",
  ],
  "sillyLogs": Array [],
  "verboseLogs": Array [],
  "warningLogs": Array [],
}
`;
